### PR TITLE
doc/rados/operations/health-checks: osd section

### PR DESCRIPTION
**Pull request by liewegas**
First paragraph: explain what the error means.

Second or later paragraph: describe steps to fix or mitigate.

Signed-off-by: Sage Weil <sage@redhat.com>

### CODE REVIEW

File: doc/rados/operations/health-checks.rst [hunk 1]
```diff
@@ -41,50 +48,176 @@ all OSDs on a host.
...
+  ceph df
 
 OSDMAP_FLAGS
 ____________
 
-
+One or more cluster flags of interest has been set.  These flags include:
+
+* *full* - the cluster is flagged as full and cannot service writes
+* *pauserd*, *pausewr*, *pauserec* - paused reads, writes, or
```
Comment (xiexingguo) [inline doc/rados/operations/health-checks.rst]: **pauserec** is not available(functionable)

File: doc/rados/operations/health-checks.rst [hunk 2]
```diff
@@ -41,124 +48,463 @@ all OSDs on a host.
...
+The OSD can be removed from the CRUSH hierarchy with::
+
+  ceph osd crush rm osd.<id>
 
 OSD_OUT_OF_ORDER_FULL
 _____________________
 
+The utilization thresholds for `backfillfull`, `nearfull`, `full`,
+and/or `failsafe_full` are not ascending.  In particular, we expect
+`backfillfull < nearful`, `nearfull < full`, and `full <
```
Comment (jdurgin) [inline doc/rados/operations/health-checks.rst]: typo: missing l in nearful

File: doc/rados/operations/health-checks.rst [hunk 3]
```diff
@@ -41,124 +48,463 @@ all OSDs on a host.
...
+
+Utilization by pool can be checked with::
+
+  ceph df
+
+The currently defined `full` ratio can be seen with::
+
+  ceph osd dump | grep full_ratio
+
+A short-term workaround restore write availability is to raise the full
```
Comment (jdurgin) [inline doc/rados/operations/health-checks.rst]: s/restore/to restore/

File: doc/rados/operations/health-checks.rst [hunk 4]
```diff
@@ -41,124 +48,463 @@ all OSDs on a host.
...
+determined by the ``mon_crush_min_required_version`` config option.
+See :doc:`/rados/operations/crush-map/#tunables` for more information.
 
 OLD_CRUSH_STRAW_CALC_VERSION
 ____________________________
 
+The CRUSH map is using an older, non-optimal method for calculating
+intermediate weight values for ``straw`` buckets.
+
+The CRUSH map should be update dot use the newer method
```
Comment (jdurgin) [inline doc/rados/operations/health-checks.rst]: s/ dot/d to/

File: doc/rados/operations/health-checks.rst [hunk 5]
```diff
@@ -41,124 +48,463 @@ all OSDs on a host.
...
 
+One or more pools has reached its quota and is no longer allowing writes.
+
+Pool quotas and utilization can be seen with::
+
+  ceph df detail
+
+You can either raise the pool quota with::
+
+  ceph osd pool set-quota <poolname> max_object <num-objects>
```
Comment (jdurgin) [inline doc/rados/operations/health-checks.rst]: 'max_object' should be plural

File: doc/rados/operations/health-checks.rst (hunk 4 above)
Comment (tchaikov) [inline doc/rados/operations/health-checks.rst]: s/use/using/

File: doc/rados/operations/health-checks.rst [hunk 6]
```diff
@@ -41,124 +48,463 @@ all OSDs on a host.
...
+more information.
 
 SMALLER_PGP_NUM
 _______________
 
+One or more pools has a ``pgp_num`` value less than ``pg_num``.  This
+is normally and indication that the PG count was increased without
+also increasing the placement behavior.
+
+This is sometime done deliberately to separate out the `split` step
```
Comment (jdurgin) [inline doc/rados/operations/health-checks.rst]: 'sometime' should be plural

File: doc/rados/operations/health-checks.rst [hunk 7]
```diff
@@ -41,124 +48,463 @@ all OSDs on a host.
...
+
+Please refer to
+:doc:`placement-groups#Choosing-the-number-of-Placement-Groups` for
+more information.
 
 SMALLER_PGP_NUM
 _______________
 
+One or more pools has a ``pgp_num`` value less than ``pg_num``.  This
+is normally and indication that the PG count was increased without
```
Comment (tchaikov) [inline doc/rados/operations/health-checks.rst]: s/and/an/

**Comment by liewegas** [comment]
fixed typos, thanks!

**Pull request merged**